### DISCUSSION TITLE
Clickhouse/Kafka error: CANNOT_PARSE_PROTOBUF_SCHEMA

### DISCUSSION THREAD

**Discussion by broonu** [akvorado/akvorado#1323] [Q&A] [answered]
I'm using Akvorado v1.11.0, Clickhouse 24.6.2.17 and Kafka 3.7.1 (no docker).
Everything was working until I tried to enable the SrcMac and DstMac filters.
After restarting orchestrator/inlet/console, I've been getting these constant errors in /var/log/clickhouse-server/clickhouse-server.err.log:

`2024.07.25 02:00:51.763846 [ 112795 ] {} <Error> StorageKafka (flows_243QR3DRGFU3LAWDAIZA7TXBYM_raw): void DB::StorageKafka::threadFunc(size_t) Code: 434. DB::Exception: Cannot parse 'flow-243QR3DRGFU3 LAWDAIZA7TXBYM.proto' file, found an error at line -1, column 0, File not found.: While executing Kafka. (CANNOT_PARSE_PROTOBUF_SCHEMA), Stack trace (when copying this message, always include the lines below):`

Also this:

`2024.07.25 02:00:51.572051 [ 112944 ] {} <Error> StorageKafka (flows_MMNUFWIF5UUS4S23FHWLPHVMW4_raw): void DB::StorageKafka::threadFunc(size_t) Code: 434. DB::Exception: Cannot parse 'flow-MMNUFWIF5UUS4S23FHWLPHVMW4.proto' file, found an error at line -1, column 0, File not found.: While executing Kafka. (CANNOT_PARSE_PROTOBUF_SCHEMA), Stack trace (when copying this message, always include the lines below):`

My Clickhouse are talking to Kafka in another Topic:
```
2024.07.25 02:03:31.302028 [ 112939 ] {} <Trace> StorageKafka (flows_LAABIGYMRYZPTGOYIIFZNYDEQM_raw): Polled batch of 1146 messages. Offsets position: [flows-LAABIGYMRYZPTGOYIIFZNYDEQM[0:2834139226]]
2024.07.25 02:03:31.311174 [ 112794 ] {} <Trace> StorageKafka (flows_LAABIGYMRYZPTGOYIIFZNYDEQM_raw): Empty assignment.
2024.07.25 02:03:31.330723 [ 112792 ] {} <Debug> StorageKafka (flows_243QR3DRGFU3LAWDAIZA7TXBYM_raw): Started streaming to 1 attached views
```

```
akvorado@akvorado:~$ sudo ls -lah /var/lib/clickhouse/format_schemas
total 12K
drwxr-x---  2 clickhouse clickhouse 4.0K Jul  8 02:25 .
drwx------ 16 clickhouse clickhouse 4.0K Jul  7 02:07 ..
-rw-r--r--  1 root       root       1.2K Jul 25 01:26 flow-LAABIGYMRYZPTGOYIIFZNYDEQM.proto

```

How can I get rid of these "flows_243QR3DRGFU3LAWDAIZA7TXBYM_raw" and "flows_MMNUFWIF5UUS4S23FHWLPHVMW4_raw" tries?

**Comment by broonu** [comment]
Well, I just dropped these tables and the errors are gone, could this break my environment?

**Reply by vincentbernat** [reply, marked as answer]
No, this should be OK. When changing the schema, ClickHouse also needs to be restarted to copy the schema from the orchestrator.

**Reply by broonu** [reply]
> No, this should be OK. When changing the schema, ClickHouse also needs to be restarted to copy the schema from the orchestrator.

Thank you Vincent. And congratulations for the Akvorado, it's really awsome!

**Comment by dbray925** [comment]
I just stumbled onto this same issue, and the same fix worked for me. However, stop, patch and reboot OS, start everything back up, and the issue comes back again. 

@broonu would you mind sharing some of your config.yaml settings with me? I'm in the exact same setup (no docker) that you are in. Only difference for me, is that _no_ data is showing up in:

```
SELECT *
FROM flows

Query id: 89749fee-33b8-418a-9ede-6614693f1695

Ok.

0 rows in set. Elapsed: 0.003 sec.
```

I must be missing an initial step, but can't seem to figure it out. Thank you!

**Reply by vincentbernat** [reply]
You need to install the schema in `/var/lib/clickhouse/format_schemas`. See the bottom of the troubleshooting section in the documentation.

**Reply by dbray925** [reply]
Ahhhh, OK, that makes sense. I'm using the pre-built binary install, so that doesn't give me anything in the format_schemas directory. It's empty. My apologies for probably missing something very simple, but I'm stumped. I guess this is exactly like: https://github.com/akvorado/akvorado/discussions/1040 but there's a missing step for binary users that I can't figure out. The documentation, to me, is a little confusing, because "curl http://akvorado/api/v0/orchestrator/clickhouse/init.sh | sh" is a dead link for me. I'm close, I can feel it, just need to understand this last piece to get ClickHouse ready. Thank you!

**Reply by vincentbernat** [reply]
With the binary, you run one instance of the inlet (`akvorado inlet`), one instance of the console `akvorado console` and one instance of the orchestrator (`akvorado orchestrator`). All of them exposes an API. You need to use the IP for the orchestrator instance and you get some kind of init script to setup Clickhouse. Something like `curl http://192.168.1.1:8081/api/v0/orchestrator/clickhouse/init.sh`.

The docker setup is easier to setup, maybe you could have a look at how it works with it before trying to not use it.

**Reply by dbray925** [reply]
Yeah, had all that:
$ sudo netstat -taunp | grep akvorado | grep -v ESTABLISHED
tcp        0      0 0.0.0.0:10179           0.0.0.0:*               LISTEN      156436/akvorado    <---- no idea
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      156343/akvorado     <---- orchestrator API
tcp        0      0 0.0.0.0:8081            0.0.0.0:*               LISTEN      156436/akvorado     <---- inlet API
tcp        0      0 0.0.0.0:8082            0.0.0.0:*               LISTEN      156452/akvorado     <---- console
udp        0      0 0.0.0.0:9995            0.0.0.0:*                           156436/akvorado        <---- inlet NetFlow

Just need a little help with trying to figure out _which_ port to point at for the init.sh, as I'm not familiar with some of these apps. Makes sense now, just had some initial confusing from the documentation. I can see now that my ClickHouse isn't properly configured. Once I got that going, then hopefully everything will fall into place. I'm close, again, just a little weird issue with ClickHouse that I hope to figure out next week when I have more time. 

Docker introduces a bunch of other issues we'd rather not deal with on an enterprise level. While great for testing apps, for us, it is not really suited for bigger installs, support, backups, monitoring, security, upgrades, reporting, logging, auditing, etc. I followed along https://omgwtfbbq.ca/2024/04/akvorado-basic-install-without-docker/ but just couldn't seem to put this last little piece together.  Thank you for your help and guidance. Really nice looking app you have, and I appreciate all your hard work!
